From 78a6bcfed4b73f13b9973afd69b76067dd4a5dde Mon Sep 17 00:00:00 2001
From: Calixte Denizet <calixte.denizet@gmail.com>
Date: Mon, 4 Oct 2021 11:07:56 +0200
Subject: [PATCH] Remove FlushViewOfFile when unmaping gcda files   - it can
 causes bad performances with slow disks;   - MS docs say that it's mainly
 useful in case of hard failures (OS crash, electrical failure, ...): so it's
 useless to call this function when ccov builds run on CI.

---
 compiler-rt/lib/profile/GCDAProfiling.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/compiler-rt/lib/profile/GCDAProfiling.c b/compiler-rt/lib/profile/GCDAProfiling.c
index 4293e8f7b5bf..83650d33c95d 100644
--- a/compiler-rt/lib/profile/GCDAProfiling.c
+++ b/compiler-rt/lib/profile/GCDAProfiling.c
@@ -286,11 +286,6 @@ static int map_file() {
 
 static void unmap_file() {
 #if defined(_WIN32)
-  if (!FlushViewOfFile(write_buffer, file_size)) {
-    fprintf(stderr, "profiling: %s: cannot flush mapped view: %lu\n", filename,
-            GetLastError());
-  }
-
   if (!UnmapViewOfFile(write_buffer)) {
     fprintf(stderr, "profiling: %s: cannot unmap mapped view: %lu\n", filename,
             GetLastError());
-- 
2.33.0

